src/mpp/Dec.hpp

```diff
@@ -190,7 +193,7 @@ constexpr bool is_path_item_static(size_t item)
 
 constexpr bool is_path_item_dynamic(size_t item)
 {
-	return path_item_type(item) >= PIT_STADYN;
+	return path_item_type(item) >= PIT_STADYN && path_item_type(item) < PIT_OPTIONAL;
```

Comment:
Just for clarification, now you made `PIT_OPTIONAL` to have 1 argument for `expected_arg_count()`, like we discussed yesterday, right? Maybe rename it to `PIT_DYN_OPTIONAL` for uniformity?

Reply:
No, optional does not have 1 argument.
This one argument is now used to remember how many elements has an array or map from MsgPack packet. Optional is not present in MsgPack, so I think that it's better to treat it as not a dynamic object.